pkg/storage/streams/store.go

```diff
@@ -96,6 +97,21 @@ func NewStreamStore(segments segments.Store, segmentSize int64, rootKey string,
 func (s *streamStore) Put(ctx context.Context, path paths.Path, data io.Reader, metadata []byte, expiration time.Time) (m Meta, err error) {
 	defer mon.Task()(&ctx)(&err)
 
+	// if previously same file uploaded, then clean up
+	lastSegmentMeta, err := s.segments.Meta(ctx, path.Prepend("l"))
+
+	// previously file uploaded
+	if err == nil {
+		msi := pb.MetaStreamInfo{}
+		err = proto.Unmarshal(lastSegmentMeta.Data, &msi)
+		if err == nil {
+			err = s.Delete(ctx, path)
+			if err != nil {
+				zap.S().Warnf("Failed to delete the references of previous upload %v", err)
```

Comment:
I wonder if we should fail the request (i.e. return the error) instead of just logging a warning. It's like we failed overwriting an existing file and this should be an error aborting the request.

Reply:
made change to abort when overwriting failed.